### PR TITLE
rbd-ggate: fix compile errors from ceph::mutex update

### PR DESCRIPTION
**Pull request by wjwithagen**
This wil fix errors like:
```
/home/jenkins/workspace/ceph-master/src/tools/rbd_ggate/Server.cc:36:14: error: no member named 'WaitInterval' in 'ceph::condition_variable_debug'
      m_cond.WaitInterval(m_lock, utime_t(1, 0));
      ~~~~~~ ^
/home/jenkins/workspace/ceph-master/src/tools/rbd_ggate/Server.cc:82:10: error: no member named 'Signal' in 'ceph::condition_variable_debug'
  m_cond.Signal();
  ~~~~~~ ^
/home/jenkins/workspace/ceph-master/src/tools/rbd_ggate/Server.cc:89:75: error: expected ';' after return statement
  m_cond.wait(locker, [this] { return !m_io_finished.empty() || m_stopping});
                                                                          ^
                                                                          ;
/home/jenkins/workspace/ceph-master/src/tools/rbd_ggate/Server.cc:109:12: error: no member named 'Wait' in 'ceph::condition_variable_debug'
    m_cond.Wait(m_lock);
    ~~~~~~ ^
```

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/tools/rbd_ggate/Server.cc [hunk 1]
```diff
@@ -31,9 +31,9 @@ void Server::run() {
   dout(20) << "entering run loop" << dendl;
 
   {
-    std::lock_guard locker{m_lock};
+    std::unique_lock locker{m_lock};
     while (!m_stopping) {
-      m_cond.WaitInterval(m_lock, utime_t(1, 0));
+      m_cond.wait_for(locker, 1s);
```
Comment (tchaikov) [inline src/tools/rbd_ggate/Server.cc]: could take this opportunity to use 
```c++
m_cond.wait_for(locker, [this] { return m_stopping; });
```

File: src/tools/rbd_ggate/Server.cc [hunk 2]
```diff
@@ -79,14 +79,14 @@ void Server::io_finish(IOContext *ctx) {
...
-  m_cond.Signal();
+  m_cond.notify_all();
 }
 
 Server::IOContext *Server::wait_io_finish() {
   dout(20) << dendl;
 
   std::unique_lock locker{m_lock};
-  m_cond.wait(locker, [this] { return !m_io_finished.empty() || m_stopping});
+  m_cond.wait(locker, [this]() { return (!m_io_finished.empty() || m_stopping); });
```
Comment (tchaikov) [inline src/tools/rbd_ggate/Server.cc]: why would you want to change this part?

File: src/tools/rbd_ggate/Server.cc [hunk 3]
```diff
@@ -103,10 +103,10 @@ void Server::wait_clean() {
 
   ceph_assert(!m_reader_thread.is_started());
 
-  std::lock_guard locker{m_lock};
+  std::unique_lock locker{m_lock};
 
   while (!m_io_pending.empty()) {
-    m_cond.Wait(m_lock);
+    m_cond.wait(locker);
```
Comment (tchaikov) [inline src/tools/rbd_ggate/Server.cc]: better off putting
```c++
m_cond.wait(locker, [this] {return m_io_pending.empty();});
```

File: src/tools/rbd_ggate/Server.cc (hunk 2 above)
Comment (wjwithagen) [reply src/tools/rbd_ggate/Server.cc]: @tchaikov 
to please Clang:
```
/home/jenkins/workspace/ceph-master/src/tools/rbd_ggate/Server.cc:89:75: error: expected ';' after return statement
  m_cond.wait(locker, [this] { return !m_io_finished.empty() || m_stopping});
                                                                          ^
                                                                          ;
1 error generated.
```

File: src/tools/rbd_ggate/Server.cc (hunk 3 above)
Comment (wjwithagen) [reply src/tools/rbd_ggate/Server.cc]: @tchaikov 
I was going for the quick fix, but fixing these makse sense as well.

**Comment by wjwithagen** [comment]
Compiles oke:
http://cephdev.digiware.nl:8180/jenkins/job/ceph-master/3604/console

File: src/tools/rbd_ggate/Server.cc [hunk 4]
```diff
@@ -31,10 +31,8 @@ void Server::run() {
   dout(20) << "entering run loop" << dendl;
 
   {
-    std::lock_guard locker{m_lock};
-    while (!m_stopping) {
-      m_cond.WaitInterval(m_lock, utime_t(1, 0));
-    }
+    std::unique_lock locker{m_lock};
+    m_cond.wait(locker, [this] { return !m_stopping;});
```
Comment (tchaikov) [inline src/tools/rbd_ggate/Server.cc]:
```suggestion
    m_cond.wait(locker, [this] { return m_stopping;});
```

File: src/tools/rbd_ggate/Server.cc [hunk 5]
```diff
@@ -103,11 +101,8 @@ void Server::wait_clean() {
 
   ceph_assert(!m_reader_thread.is_started());
 
-  std::lock_guard locker{m_lock};
-
-  while (!m_io_pending.empty()) {
-    m_cond.Wait(m_lock);
-  }
+  std::unique_lock locker{m_lock};
+  m_cond.wait(locker, [this] { return !m_io_pending.empty();});
```
Comment (tchaikov) [inline src/tools/rbd_ggate/Server.cc]:
```suggestion
  m_cond.wait(locker, [this] { return m_io_pending.empty();});
```

Comment (wjwithagen) [reply src/tools/rbd_ggate/Server.cc]: Copy Paste errors...
Wil fix

**Pull request merged**